src/Controls/src/Core/Shell/ShellItem.cs

```diff
@@ -270,7 +270,7 @@ void OnVisibleChildRemoved(Element child)
 			if (CurrentItem == child)
 			{
 				if (ShellItemController.GetItems().Count == 0)
-					ClearValue(CurrentItemProperty, specificity: SetterSpecificity.FromHandler);
+					ClearValue(CurrentItemProperty);
```

Comment:
what's the rationale for dropping the specificity ?

Reply:
The rationale for dropping the specificity in this case is due to how values are set and managed. Initially, **ClearValue(CurrentItemProperty, SetterSpecificity.FromHandler)** was used to clear values set by a handler, such as FromHandler.

When a value is set manually or bound using SetValue, the specificity FromHandler is automatically replaced by ManualValueSetter, making the FromHandler value irrelevant. The existing logic in [ClearValueCore](https://github.com/dotnet/maui/blob/87253a5cfc6619237503c62b225b9e10561dedad/src/Controls/src/Core/BindableObject.cs#L120)  retrieves the current setter key (e.g., ManualValueSetter or other), but if the value was manually set, passing FromHandler to ClearValue becomes redundant as this key is no longer present in the context values. 

If FromHandler has not been overridden by another setter, the method clears the value based on the original key retrieved from the property. 

https://github.com/dotnet/maui/blob/87253a5cfc6619237503c62b225b9e10561dedad/src/Controls/src/Core/BindableObject.cs#L128-L129

If it has been overridden, the method automatically clears the value according to the current setter, ensuring that the value is cleared appropriately.

https://github.com/dotnet/maui/blob/87253a5cfc6619237503c62b225b9e10561dedad/src/Controls/src/Core/BindableObject.cs#L140 

It is not necessary to pass the FromHandler in the ClearValue. Could you please share your thoughts on this approach?

Reply:
of you clear the value set FromHandler, it should go back to the previous value with the highest specificity. isn't this what you want ?

Reply:
Since we are passing the FromHandler specificity, it only checks the values set by the handler. Therefore, the manually set value for the CurrentItem property is not cleared when setting visibility. Resetting to the highest specificity is not our intention. Instead, we clear the value without specifying specificity to ensure it is properly cleared.